stock_valuation_fifo_lot/models/stock_move.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2022 Ecosoft Co., Ltd (https://ecosoft.co.th)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _prepare_common_svl_vals(self):
+        """
+        Prepare lots/serial numbers on stock valuation report
+        """
+        self.ensure_one()
+        res = super()._prepare_common_svl_vals()
+        res.update(
+            {
+                "lot_ids": [(6, 0, self.lot_ids.ids)],
+            }
+        )
+        return res
+
+    def _create_out_svl(self, forced_quantity=None):
+        """
+        Send context current move to _create_out_svl function
+        """
+        layers = self.env["stock.valuation.layer"]
+        for move in self:
+            move = move.with_context(used_in_move_id=move.id)
+            layer = super(StockMove, move)._create_out_svl(
+                forced_quantity=forced_quantity
+            )
+            layers |= layer
+        return layers
+
+    def _create_in_svl(self, forced_quantity=None):
+        """
+        1. Check stock move - Multiple lot on the stock move is not
+           allowed for incoming transfer
+        2. Change product standard price to first available lot price
+        """
+        layers = self.env["stock.valuation.layer"]
+        for move in self:
+            if len(move.lot_ids) > 1:
+                raise UserError(
+                    _(
+                        "Multiple lot on the stock move is not allowed for incoming transfer"
+                    )
+                )
```

Comment:
The scenario to have more than 1 lot per move is common. So we should reconsider this check.
```suggestion
```

Reply:
if not check, it affect with accounting or cost about it don't know how to accounting which lot

Reply:
IIUC, the main purpose of the method is to find first lot that has transferred into current company and use its price as cost. So we can assume that all lots of the same delivery has same price for same product. Then IMHO current logic (`min` + `mapped`) is enough to deal with multiple lot on the stock move.

Reply:
Thanks for the work anyway. :)